test_util/Makefile

```diff
@@ -80,6 +86,39 @@ clean:
 	rm -rf ./*.pem
 	rm -rf ./*.pub
 
+
+################################# DISGNOSTICS #################################
+
+diagnostics_api = localhost:1050/system/health/v1/diagnostics
+ifneq ($(TF_VAR_variant), open)
+	diagnostics_api = --unix-socket /var/run/dcos/dcos-diagnostics.sock http:/system/health/v1/diagnostics
+endif
+
+# Start diagnostics bundle creation.
+diagnostics-%.started: cluster.json
+	@echo "##teamcity[blockOpened name='diagnostics-bundle' description='Download Diagnostics bundle']"
+	ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i $(SSH_KEY) centos@$(shell jq '.masters_public_ip.value' $<) -- \
+		"curl -X PUT $(diagnostics_api)/$*"
+	touch $@
+
+# Wait until diagnostics bundle is ready.
+diagnostics-%.done: cluster.json diagnostics-%.started 
+	ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i $(SSH_KEY) centos@$(shell jq '.masters_public_ip.value' $<) -- \
+		"curl -s $(diagnostics_api)/$* -o status.json; \
+		until jq '.status' status.json | grep -q 'Done'; do cat status.json; curl -s $(diagnostics_api)/$* -o status.json; sleep 5; done;"
```

Comment:
How about moving loop outside of SSH to not keep open connection for a long time.

Reply:
I'm not really sure how to do that. Has `ssh` the exit code as the subcommand?